laplace/utils/matrix.py

```diff
@@ -118,7 +109,14 @@ def decompose(self, damping=False):
         for F in self.kfacs:
             Qs, ls = list(), list()
             for Hi in F:
-                l, Q = symeig(Hi)
+                if Hi.ndim > 1:
+                    # Dense Kronecker factor.
+                    l, Q = symeig(Hi)
+                else:
+                    # Diagonal Kronecker factor.
+                    l = Hi
+                    # This might be too memory intensive since len(Hi) can be large.
```

Comment:
Remove this comment that is from sorting previously?

Reply:
No it is due to the `torch.eye`, since the Kronecker factor is diagonal it might be too large to explicitly build a `len(H_i) x len(H_i)` matrix.